internal/cli/cmd/auth/utils/utils.go

```diff
@@ -0,0 +1,101 @@
+/*
+Copyright (C) 2022-2023 ApeCloud Co., Ltd
+
+This file is part of KubeBlocks project
+
+This program is free software: you can redistribute it and/or modify
+it under the terms of the GNU Affero General Public License as published by
+the Free Software Foundation, either version 3 of the License, or
+(at your option) any later version.
+
+This program is distributed in the hope that it will be useful
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU Affero General Public License for more details.
+
+You should have received a copy of the GNU Affero General Public License
+along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+package utils
+
+import (
+	"context"
+	"net/http"
+	"net/url"
+	"os"
+	"os/exec"
+	"strings"
+
+	"github.com/fatih/color"
+	"github.com/mattn/go-isatty"
+)
+
+func IsTTY() bool {
+	return isatty.IsTerminal(os.Stdout.Fd()) || isatty.IsCygwinTerminal(os.Stdout.Fd())
+}
+
+// OpenBrowser opens a web browser at the specified url.
+func OpenBrowser(goos, url string) *exec.Cmd {
+	if !IsTTY() {
+		panic("OpenBrowser called without a TTY")
+	}
+
+	exe := "open"
+	var args []string
+	switch goos {
+	case "darwin":
+		args = append(args, url)
+	case "windows":
+		exe, _ = exec.LookPath("cmd")
+		r := strings.NewReplacer("&", "^&")
+		args = append(args, "/c", "start", r.Replace(url))
+	default:
+		exe = linuxExe()
+		args = append(args, url)
+	}
+
+	cmd := exec.Command(exe, args...)
+	cmd.Stderr = os.Stderr
+	return cmd
+}
+
+func linuxExe() string {
+	exe := "xdg-open"
+
+	_, err := exec.LookPath(exe)
+	if err != nil {
+		_, err := exec.LookPath("wslview")
+		if err == nil {
+			exe = "wslview"
+		}
+	}
+
+	return exe
+}
+
+func NewRequest(ctx context.Context, url string, payload url.Values) (*http.Request, error) {
+	req, err := http.NewRequestWithContext(
+		ctx,
+		http.MethodPost,
+		url,
+		strings.NewReader(payload.Encode()),
+	)
+	if err != nil {
+		return nil, err
+	}
+	req.Header.Set("Content-Type", "application/x-www-form-urlencoded")
+	req.Header.Set("Accept", "application/json")
+	return req, nil
+}
+
+// BoldRed returns a string formatted with red and bold.
+func BoldRed(msg interface{}) string {
+	return color.New(color.FgRed).Add(color.Bold).Sprint(msg)
+}
+
+// Bold returns a string formatted with bold.
+func Bold(msg interface{}) string {
```

Comment:
Better to use `printer.BoldRed`, ignore duplicate.

Reply:
LGTM